### PR TITLE
Fix a bug of benchmark_app for binary input filling.

### PR DESCRIPTION
**Pull request by jade-cho**
Some input tensors used as parameter are not reshaped when the batch is N. In this case, it tries to allocate the binary input N times and then it is aborted. This PR fixes this issue.

### CODE REVIEW

File: samples/cpp/benchmark_app/inputs_filling.cpp [hunk 1]
```diff
@@ -117,6 +117,20 @@ void fillBlobBinary(Blob::Ptr& inputBlob,
...
+    auto tensorDesc = inputBlob->getTensorDesc();
+    ss << tensorDesc.getLayout();
+    auto layout = ss.str();
+    std::size_t batchIndex = layout.find("N");
+    if (batchIndex == std::string::npos) {
+        adjBatchSize = 1;
+    } else if (tensorDesc.getDims().at(batchIndex) != batchSize) {
+        adjBatchSize = tensorDesc.getDims().at(batchIndex);
+    }
+
```
Comment (fzhar) [inline samples/cpp/benchmark_app/inputs_filling.cpp:133]: Shouldn't we reshape network using new provided batch size instead ?

Comment (jade-cho) [reply samples/cpp/benchmark_app/inputs_filling.cpp:133]: It is not reshaped due to the following code.
samples/cpp/benchmark_app/utils.hpp:111 ~ 118
        // Update shape with batch if needed
        if (batch_size != 0) {
            std::size_t batch_index = info.layout.find("N");
            if ((batch_index != std::string::npos) && (info.shape.at(batch_index) != batch_size)) {
                info.shape[batch_index] = batch_size;
                reshape_required = true;
            }
        }
Its layout is only 'C'. Do you think we should modify this code to reshape 'C' layout tensor to 'NC'?

**Comment by ivikhrev** [comment]
@jade-cho could you please provide with what command line parameters you faced this issue? Because we are preparing big update for benhmark app (#7788, #8609) , and we changed part of this logic there. I want to test and check if there still a bug.

**Comment by jade-cho** [comment]
I mailed you. As mentioned in the mail, if there are no problems, could you please merge this MR? I want to periodically test the accuracy of the model for various batches in the latest version of openvino.

**Comment by fzhar** [comment]
@jade-cho, we can merge this, but these changes wil be overwritten by upcoming update (that is going to be merged really soon). To make sure that this issue is covered, could you provide exact problem description: command line that causes problem, current behavior and expected behavior ?

**Comment by ivikhrev** [comment]
@jade-cho possible solution to your problem is to pass `-layout` argument with `NC` value for inputs which layouts is `C` ( I mailed you more detail info).

**Comment by jade-cho** [comment]
@fzhar @ivikhrev I mailed details with log messages. -layout option makes different issue on GPU device. And -layout option didn't reshape C tensor to NC tensor. In gpu_plugin, the input format could affect graph compilation. So, using the -layout option seems to be insufficient to solve this problem. Please merge this PR. After your updates merged, I will follow up.

**Review by fzhar — APPROVED** [review]

**Pull request merged**